pkg/virt-controller/watch/vm.go

```diff
@@ -2736,6 +2798,51 @@ func (c *VMController) hasOrdinalNetworkInterfaces(vmi *virtv1.VirtualMachineIns
 	return hasOrdinalIfaces, nil
 }
 
+func (c *VMController) handleMemoryHotplugRequest(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if vmi == nil || vmi.DeletionTimestamp != nil {
+		return nil
+	}
+
+	if vm.Spec.LiveUpdateFeatures == nil || vm.Spec.LiveUpdateFeatures.Memory == nil {
+		return nil
+	}
+
+	if vm.Spec.Template.Spec.Domain.Memory == nil || vmi.Spec.Domain.Memory == nil {
+		return nil
+	}
+
+	guestMemory := vmi.Spec.Domain.Memory.Guest
+
+	if vmi.Status.Memory != nil &&
+		vmi.Status.Memory.GuestCurrent != nil &&
+		!vm.Spec.Template.Spec.Domain.Memory.Guest.Equal(*guestMemory) {
+
+		conditionManager := controller.NewVirtualMachineInstanceConditionManager()
+
+		if conditionManager.HasConditionWithStatus(vmi,
+			virtv1.VirtualMachineInstanceMemoryChange, k8score.ConditionTrue) {
+			return fmt.Errorf("another memory hotplug is in progress")
+		}
+
+		if migrations.IsMigrating(vmi) {
+			return fmt.Errorf("memory hotplug is not allowed while VMI is migrating")
+		}
+
+		updateGuest := fmt.Sprintf(`{ "op": "replace", "path": "/spec/domain/memory/guest", "value": "%s"}`, vm.Spec.Template.Spec.Domain.Memory.Guest.String())
```

Comment:
Why won't you go with the test & replace approach here?

Reply:
Definitely better, fixed.